File: face-check/ApiModule/src/main/java/com/zikpak/facecheck/taxesServices/services/LocationTrackingGoogle/LocationUpdateDto.java
package com.zikpak.facecheck.taxesServices.services.LocationTrackingGoogle;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.Instant;

@Data
public class LocationUpdateDto {

    @NotNull
    private Double latitude;

    @NotNull
    private Double longitude;

    @NotNull
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", timezone = "UTC")
    private Instant timestamp;

    private Double accuracy;
    private Double speed;
    private Double bearing;
    private Double altitude;
    private String provider;
    private Integer batteryLevel;


}

File: face_check/lib/models/location_update_dto.dart
class LocationUpdateDto {
  final double latitude;
  final double longitude;
  final DateTime timestamp;
  final double? accuracy;
  final double? speed;
  final double? bearing;
  final double? altitude;
  final String? provider;
  final int? batteryLevel;

  LocationUpdateDto({
    required this.latitude,
    required this.longitude,
    required this.timestamp,
    this.accuracy,
    this.speed,
    this.bearing,
    this.altitude,
    this.provider,
    this.batteryLevel,
  });

  Map<String, dynamic> toJson() {
    return {
      'latitude': latitude,
      'longitude': longitude,
      'timestamp': timestamp.toUtc().toIso8601String(),
      'accuracy': accuracy,
      'speed': speed,
      'bearing': bearing,
      'altitude': altitude,
      'provider': provider,
      'batteryLevel': batteryLevel,
    };
  }

  factory LocationUpdateDto.fromJson(Map<String, dynamic> json) {
    return LocationUpdateDto(
      latitude: json['latitude']?.toDouble() ?? 0.0,
      longitude: json['longitude']?.toDouble() ?? 0.0,
      timestamp: DateTime.parse(json['timestamp']),
      accuracy: json['accuracy']?.toDouble(),
      speed: json['speed']?.toDouble(),
      bearing: json['bearing']?.toDouble(),
      altitude: json['altitude']?.toDouble(),
      provider: json['provider'],
      batteryLevel: json['batteryLevel']?.toInt(),
    );
  }
}